release_dashboard/lib/widgets/cherrypicks_substeps.dart

```diff
@@ -0,0 +1,169 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:conductor_core/conductor_core.dart';
+import 'package:conductor_core/proto.dart' as pb;
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import '../logic/repositories_str.dart';
+import '../models/cherrypick.dart';
+import '../models/conductor_status.dart';
+import '../models/repositories.dart';
+import '../state/status_state.dart';
+import 'common/checkbox_substep.dart';
+import 'common/url_button.dart';
+
+enum CherrypicksSubstep {
+  verifyRelease,
+  applyCherrypicks,
+}
+
+/// Group and display all substeps related to Apply Engine/Framework Cherrypicks step into a widget.
+///
+/// When all substeps are completed, [nextStep] can be executed to proceed to the next step.
+class CherrypicksSubsteps extends StatefulWidget {
+  const CherrypicksSubsteps({
+    Key? key,
+    required this.nextStep,
+    required this.repository,
+  }) : super(key: key);
+
+  final VoidCallback nextStep;
+  final Repositories repository;
+
+  @override
+  State<CherrypicksSubsteps> createState() => CherrypicksSubstepsState();
+
+  static const Map<CherrypicksSubstep, String> substepTitles = <CherrypicksSubstep, String>{
+    CherrypicksSubstep.verifyRelease: 'Verify the Release Number',
+    CherrypicksSubstep.applyCherrypicks: 'Apply cherrypicks and resolve conflicts',
+  };
+
+  static Map<pb.CherrypickState, String> cherrypickStates = <pb.CherrypickState, String>{
```

Comment:
This is a good use case for [extension methods](https://dart.dev/guides/language/extension-methods). We could make an extension on `pb.CherrypickState` for `toString` and import that here.

Reply:
Created an issue here: https://github.com/flutter/flutter/issues/94387, and added a todo here.

Reply:
Let's add the extension methods for this and `repositoryName` until those changes roll in

Reply:
Done, please take another look.